python/tabby/tabby-data-pipeline/modal/predict.py

```diff
@@ -157,45 +160,60 @@ async def main(language, file):
     print(health_resp)
     assert(health_resp['model'] == MODEL_ID)
 
-    whole_path_file = "./data/" + MODEL_ID.split("/")[-1] + "/" + language + "/" + file
-
-    if file == 'line_completion.jsonl':
-        crossfile_context = False
-    else:
-        crossfile_context = True
-
-    objs = []
-    with open(whole_path_file) as fin:
-        for line in fin:
-            obj = json.loads(line)
-            objs.append(obj)
-
-    df = pd.DataFrame(objs)
-
-    outputs = await asyncio.gather(*[model.complete.remote.aio(language, crossfile_context, index, row) for index, row in df.iterrows()])
-
-    skipped = 0
-    success = 0
-    error = 0
-
-    for index, prediction, error_msg in outputs:
-        if index is None:
-            skipped += 1
-        elif prediction is not None:
-            df.loc[index, 'prediction'] = prediction
-            success += 1
-        else:
-            df.loc[index, 'error'] = error_msg
-            error += 1
-    print(f"Skipped {skipped} rows, {success} rows with predictions, {error} rows with errors")
-
-    with open(whole_path_file, 'w') as fout:
-        for index, row in df.iterrows():
-            json.dump(row.to_dict(), fout)
-            fout.write('\n')
-                
-
-    with open("log.txt", "a") as flog:
-        flog.write(f"model: {MODEL_ID}; language: {language}; file: {file}")
-        flog.write(f"Skipped {skipped} rows, {success} rows with predictions, {error} rows with errors")
-        flog.write("\n\n")
\ No newline at end of file
+    
+    for file in ['line_completion.jsonl', 'line_completion_rg1_bm25.jsonl', 'line_completion_oracle_bm25.jsonl']:
```

Comment:
please extract function for this, e.g `read_pandas_frame`

Reply:
I'm not sure what this meant? extract function for the "for" loop? or do you mean extract function for reading all the three files?